### PR TITLE
fix: internal error creating a cron job

### PR DESCRIPTION
**Pull request by sr-leann**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Is there any issue related to this PR in other repository?** (such as dojot/dojot)
*dojot/dojot#2325

* **Other information**:

### CODE REVIEW

**Review by rodrigoaasm — DISMISSED** [review]
Great job!

File: app/cron.js [hunk 1]
```diff
@@ -290,6 +297,15 @@ class CronManager {
         'Communication with database (mongoDB) was established.'
       );
 
+      const api = CronManager.createAxios();
+      const tenants = await api
+        .get('http://auth:5000/admin/tenants')
```
Comment (cfrancisco) [inline app/cron.js]: A hard-coded configuration doesn't sound well.

Comment (cfrancisco) [inline app/cron.js]: A hard-coded configuration doesn't sound well.

File: app/cron.js [hunk 2]
```diff
@@ -69,7 +76,7 @@ class CronManager {
           const cronJobSetPromises = [];
           for (const job of jobs) {
             cronJobSetPromises.push(
-              this._setCronJob(tenant, job.jobId, job.spec)
+              this.setCronJob(tenant, job.jobId, job.spec)
```
Comment (cfrancisco) [inline app/cron.js:81]: Split this function into two methods if possible (following the Single Responsibility Principle, google it kkk)

File: app/cron.js [hunk 3]
```diff
@@ -290,6 +297,15 @@ class CronManager {
         'Communication with database (mongoDB) was established.'
       );
 
+      const api = CronManager.createAxios();
+      const tenants = await api
+        .get('http://auth:5000/admin/tenants')
+        .then((response) => response.data.tenants);
+
+      for (const tenant of tenants) {
+        this.setTenant(tenant);
```
Comment (cfrancisco) [inline app/cron.js]: Hm... naming a function 'SetTenant' doesn't explain very well what this function does, maybe createTenantManagment... I don't know, but I would think about it rs.

**Review by cfrancisco — CHANGES_REQUESTED** [review]
Good job!

**Review by luissajunior — DISMISSED** [review]

File: app/cron.js (hunk 3 above)
Comment (sr-leann) [reply app/cron.js]: Done.

File: app/cron.js (hunk 2 above)
Comment (sr-leann) [reply app/cron.js:81]: Vou de português aqui:
Bem, esse é um serviço experimental e este é só um dos problemas a serem tratados, está prometido um ticket especifico para resolver essa e outras questões, como testes unitários.

File: app/cron.js (hunk 1 above)
Comment (sr-leann) [reply app/cron.js]: Done.

Comment (sr-leann) [reply app/cron.js]: Done.

File: server.js [hunk 4]
```diff
@@ -28,6 +29,8 @@ const serviceStateManager = new ServiceStateManager({
   lightship: transformObjectKeys(config.lightship, camelCase),
 });
 
+const tenantService = new TenantService(config.url.tenants);
```
Comment (cfrancisco) [inline server.js:32]: Good job!

**Review by cfrancisco — APPROVED** [review]

**Pull request closed without merging**